code/modules/vehicles/_vehicle.dm

```diff
@@ -148,3 +148,13 @@
 	if(trailer && .)
 		var/dir_to_move = get_dir(trailer.loc, newloc)
 		step(trailer, dir_to_move)
+
+// handles vehicles without riding components such as mechs bumping into things and having their access checked
+/obj/vehicle/Bump(atom/A)
+	. = ..()
+	if(GetComponent(/datum/component/riding)) // if we're a ridable vehicle, this is already handled in the riding component
```

Comment:
I have to do this with edible component still but it'd be more performant if you make this into a trait rather than using getcomponent